src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -358,9 +357,9 @@ static void wm_agent_upgrade_start_upgrades(cJSON *json_response, const cJSON* t
                 status_json = wm_agent_upgrade_send_single_task(WM_UPGRADE_AGENT_UPDATE_STATUS, agent_task->agent_info->agent_id, task_statuses[WM_TASK_FAILED]);
             }
 
-            wm_agent_upgrade_validate_task_status_message(status_json, NULL);
+            wm_agent_upgrade_validate_task_status_message(status_json, NULL, &agent_id);
 
-            wm_agent_upgrade_remove_entry(atoi(agent_key));
```

Comment:
I think is better not to change this because the status_json could be malformed and it won't return the agent_id.

Reply:
You are right, I will revert this change